README.md

```diff
@@ -185,6 +185,23 @@ describe 'myclass' do
 end
 ```
 
+```ruby
+require 'spec_helper'
+
+describe_on_supported_os 'myclass' do
+  it { is_expected.to compile.with_all_deps }
+  ...
+
+  # If you need any to specify any operating system specific tests
+  case metadata[:os_facts][:osfamily]
```

Comment:
The current setup only symbolizes part of the data used in blocks like this. How would one expanded to use non-legacy facts look? It looks like this today, for context:

```ruby
case os_facts[:os]['family']
when 'RedHat'
  ...
when 'Debian'
  ...
else
  ...
end
```

Reply:
`metadata[:os_facts]` is `os_facts` so `metadata[:os_facts][:os]['family']`.

Again, really unrelated but yes, it's really confusing that it's not fully using symbols. The root cause is [here](https://github.com/voxpupuli/facterdb/blob/208d18a91adb628c1f7b71fa6fcc68c3dcc127d5/lib/facterdb.rb#L131-L144). Ideally speaking it would use something like [HashWithIndifferentAccess](https://github.com/rails/rails/blob/main/activesupport/lib/active_support/hash_with_indifferent_access.rb) but pulling in ActiveSupport just for that feels very heavy. However, facts from FacterDB should really be read-only besides overrides. We can probably easily subclass `Hash` and override `[](key)` to call `.to_s(key)`. While we're at it, implement `override` in it as well.